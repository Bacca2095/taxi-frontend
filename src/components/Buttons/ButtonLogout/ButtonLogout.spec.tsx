import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import { ButtonLogout } from '.';

describe('ButtonLogout tests', () => {
  let onClick: jest.Mock;

  beforeEach(() => {
    onClick = jest.fn();
  });

  it('should match snapshot', () => {
    const { container } = render(<ButtonLogout onClick={onClick} />);

    expect(container).toMatchSnapshot();
  });

  it('should click on logout', async () => {
    const { findByTestId } = render(<ButtonLogout onClick={onClick} />);

    const logoutButton = await findByTestId('logout');

    fireEvent.click(logoutButton);

    expect(onClick).toBeCalled();
  });
});
