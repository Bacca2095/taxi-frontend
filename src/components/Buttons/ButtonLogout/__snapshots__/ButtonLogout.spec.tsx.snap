// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonLogout tests should match snapshot 1`] = `
<div>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-colorInherit"
    data-testid="logout"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiButton-label"
    >
      Salir
    </span>
    <span
      class="MuiTouchRipple-root"
    />
  </button>
</div>
`;
